### PR TITLE
Mss 939/export subscription data 2

### PR DESCRIPTION
**Pull request by NathanielBennett**
Not ready for merging yet. This exports a dynamo table to s3 using emr and hive. The tables we will eventually want to export don't exist yet, so I've been testing with the SFL tables. 

Couple of things:

1. I couldn't get riffraff to do an s3 deployment and a lambda deployment for the same sub-project so I've added the hive-scriptts in a top level folder and passed them to riffraff as artifacts. This means the scripts for both platforms always get published. There is a script for each platform because: 
2. I couldn't a parameterized hive script(to which I could pass 'Stage') to work as described here: https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-hive-differences.html. And I spent a day and a half trying

### CODE REVIEW

File: cloudformation.yaml [hunk 1]
```diff
@@ -124,6 +133,42 @@ Resources:
               Action: [ "kms:GenerateDataKey", "kms:Decrypt" ]
               Resource:
                 - !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/0215d06c-81c4-4896-a5da-c818770ea8db
+        - PolicyName: mobile-subscriptions-export-s3
+          PolicyDocument:
+            Statement:
+              Action:
+                - s3:GetObject
+              Effect: Allow
+              Resource: !Sub arn:aws:s3:::mobile-dist/${Stage}/mobile-subscription-export-hive-scripts/*
```
Comment (alexduf) [inline cloudformation.yaml]:
```suggestion
              Resource: !Sub arn:aws:s3:::${DeployBucket}/${Stage}/mobile-subscription-export-hive-scripts/*
```

File: cloudformation.yaml [hunk 2]
```diff
@@ -124,6 +133,42 @@ Resources:
...
+        - PolicyName: mobile-subscriptions-read-jar
+          PolicyDocument:
+            Statement:
+              Action:
+                - s3:GetObject
+              Effect: Allow
+              Resource: !Sub arn:aws:s3:::mobile-dist/${Stage}/mobile-purchases/subscription-export/*
```
Comment (alexduf) [inline cloudformation.yaml]:
```suggestion
              Resource: !Sub arn:aws:s3:::${DeployBucket}/${Stage}/mobile-purchases/subscription-export/*
```

File: cloudformation.yaml [hunk 3]
```diff
@@ -589,4 +634,39 @@ Resources:
...
+        S3Key: !Sub ${Stack}/${Stage}/${App}-subscription-export/${App}-subscription-export.jar
+      Environment:
+        Variables:
+          App: !Sub ${App}-subscription-export
+          Stack: !Ref Stack
+          Stage: !Ref Stage
+      Description: exports subscription data from dynamo db to the data lake
+      Handler: com.gu.mobile.subscription.export.Lambda::handler
+      MemorySize: 512
+      Timeout: 900
```
Comment (alexduf) [inline cloudformation.yaml]: How long does the lambda usually take? I suspect this might reasonably be around 60s rather than 900

Comment (NathanielBennett) [reply cloudformation.yaml]: Good call. The lambda quits before the emr cluster does it stuff and I forgot to go back and change this.

**Comment by alexduf** [comment]
Closed in favour of https://github.com/guardian/mobile-purchases/pull/84

**Pull request closed without merging**